src/NuGet.Clients/NuGet.PackageManagement.UI/DisplayVersion.cs

```diff
@@ -48,22 +51,28 @@ public class DisplayVersion
             Version = range.MinVersion;
             IsCurrentInstalled = isCurrentInstalled;
             AutoReferenced = autoReferenced;
+            IsDeprecated = isDeprecated;
 
             // Display a single version if the range is locked
-            if (range.HasLowerAndUpperBounds && range.MinVersion == range.MaxVersion)
-            {
-                var formattedVersionString = Version.ToString(versionFormat, VersionFormatter.Instance);
+            // If range is unlocked, display it and use the original value for floating ranges
+            var versionString = range.HasLowerAndUpperBounds && range.MinVersion == range.MaxVersion ?
+                Version.ToString(versionFormat, VersionFormatter.Instance)
+                : Range.OriginalString;
 
-                _toString = string.IsNullOrEmpty(AdditionalInfo) ?
-                    formattedVersionString :
-                    AdditionalInfo + " " + formattedVersionString;
+            _toString = "";
+            if (!string.IsNullOrEmpty(AdditionalInfo))
+            {
+                _toString += AdditionalInfo + " ";
             }
-            else
+
+            _toString += versionString;
```

Comment:
I'm not a fan of multiple string concatenations. Ok, there's only between 1 and 4 concatenations depending on the code path taken, and with this few string concat usually has better perf than creating a new string builder and using it. But I'm sure there can be better performing code while still being easy to read.

How about something like:

```csharp
string formatString;
if (!string.IsNullOrEmpty(AdditionalInfo))
{
  formatString = IsDeprecated
    ? "{0} {1}   ({2})"
    : "{0} {1}";
}
else
{
  formatString = IsDeprecated
    ? "{1}   ({2})"
    : "{1}";
}

_toString = string.Format(formatString, AdditionalInfo, versionString, Resources.Label_Deprecated);
```

although given there's only 4 formats, maybe don't use `formatString`, and just assign to `_toString` in a single `String.Concat`.

Reply:
Since those strings are localized, consider creating a string resource for each case. That makes the code easier to debug and it also makes easier to avoid spelling errors on the UI.

Reply:
The format strings are not localised, and the other strings are already localised. I don't understand which strings you're proposing should be in the resources.

Reply:
I agree, but due to @nkolev92's concerns I'm going to simplify the amount of changes here to avoid push-back.